python/lsst/rapid/analysis/visitCheck.py

```diff
@@ -0,0 +1,190 @@
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+__all__ = ['CheckVisitTask',
+           'CheckVisitTaskConfig',
+           ]
+
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+# import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.log as lsstLog
+from lsst.cp.pipe.utils import countMaskedPixels
+
+from lsst.ip.isr import IsrTask, AssembleCcdTask
+from astro_metadata_translator import ObservationInfo
+
+
+class BasicTestTask(pipeBase.Task):
+    logger = lsstLog.getLogger(' ')  # empty string results in name being "root"
+
+    def computeImageStats(self, exposure):
+        opts = afwMath.MEAN | afwMath.STDEV | afwMath.MEANCLIP | afwMath.STDEVCLIP
+        stats = afwMath.makeStatistics(exposure.maskedImage, opts)
+        mean = stats.getValue(afwMath.MEAN)
+        std = stats.getValue(afwMath.STDEV)
+        meanClip = stats.getValue(afwMath.MEANCLIP)
+        stdClip = stats.getValue(afwMath.STDEVCLIP)
+
+        # add min, max, percentiles xxx
+
+        nBad = countMaskedPixels(exposure, 'BAD')
+        msgs = []
+        msgs.append(f"Mean         = {mean:.2f}")
+        msgs.append(f"Clipped mean = {meanClip:.2f}")
+        msgs.append(f"Std dev      = {std:.2f}")
+        msgs.append(f"Clipped std  = {stdClip:.2f}")
+        msgs.append(f"nBad pix     = {nBad}")
+
+        for msg in msgs:
+            self.logger.info(msg)
+
+
+class BiasTestConfig(pexConfig.Config):
+    someOption = pexConfig.Field(
+        doc="An option",
+        dtype=bool,
+        default=True,
+    )
+
+
+class BiasTestTask(BasicTestTask):
+    ConfigClass = BiasTestConfig
+
+    def runDataRef(self, dataRef, exposure):
+        bias = dataRef.get('bias')
+        detector = exposure.getDetector()
+
+        # sanity checking - these should NEVER fail
+        assert bias.getDetector().getName() == detector.getName()
+        assert bias.getDetector().getId() == detector.getId()
+
+        if bias.image.array.shape != exposure.image.array.shape:
+            raise RuntimeError("Bias and exposure are different shapes")
+
+        # import ipdb as pdb; pdb.set_trace()
```

Comment:
stick this in a lsst debug statement, so it can be run with just the appropriate debug config?